### PR TITLE
feat: Implement wish deposit

### PR DESCRIPTION
**Pull request by wolfenrain**
## Description

Implemented the wishes API abstraction and models.

Tests are based on [the example tests](https://github.com/felangel/bloc/blob/master/examples/flutter_todos/packages/todos_api/test/) with a few minor changes.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/wishes_api/lib/src/models/wish.dart [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+class Wish extends Equatable {
+  /// {@macro wish}
+  Wish({
+    String? id,
+    required this.title,
+    this.description = '',
+    this.isFulfilled = false,
+  })  : assert(
+          id == null || id.isNotEmpty,
+          'id can not be null and should be empty',
```
Comment (felangel) [inline packages/wishes_api/lib/src/models/wish.dart]:
```suggestion
          'id cannot be null and should be empty',
```

**Review by felangel — COMMENTED** [review]
Looks great so far 💯

File: packages/wishes_api/lib/src/models/wish.dart (hunk 1 above)
Comment (wolfenrain) [reply packages/wishes_api/lib/src/models/wish.dart]: Seems that has to be changed on the tutorial as well then

File: packages/wish_deposit/lib/src/wish_deposit.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import 'package:deposit/deposit.dart';
+import 'package:wish_deposit/wish_deposit.dart';
+
+/// {@template wish_deposit}
+/// A deposit that talks to the wishes data backend.
+/// {@endtemplate}
+class WishDeposit extends Deposit<WishEntity, int> {
+  /// {@macro wish_deposit}
+  WishDeposit({
+    required DepositAdapter<int> adapter,
```
Comment (felangel) [inline packages/wish_deposit/lib/src/wish_deposit.dart:10]: Will this adapter be implemented in a separate pull request or will the project just rely on in the MemoryDepositAdapter?

Comment (wolfenrain) [reply packages/wish_deposit/lib/src/wish_deposit.dart:10]: I am planning on using supabase for the whole project, so instead of defining a constructor that can initialise a supabase adapter I thought it might be more better to just pass it to the deposit. That way I can define a single adapter on the app project and pass it along to any deposit that will need it.

File: pubspec.yaml [hunk 3]
```diff
@@ -14,12 +14,8 @@ dependencies:
   flutter_localizations:
     sdk: flutter
   intl: ^0.17.0
-  local_storage_wishes_api:
-    path: packages/local_storage_wishes_api
-  wishes_api:
-    path: packages/wishes_api
-  wishes_repository:
-    path: packages/wishes_repository
+  wish_deposit:
```
Comment (felangel) [inline pubspec.yaml:17]: can we add a ci workflow for this package?

Comment (wolfenrain) [reply pubspec.yaml:17]: Depends, what do you exactly mean with that 😄

Comment (felangel) [reply pubspec.yaml:17]: haha can we create a github workflow in the `.github/workflows/wish_deposit.yaml` to use `very_good_workflows` so that we run the standard checks whenever changes are made to `package:wish_deposit`?

File: .github/workflows/wish_deposit.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: WishDeposit Workflow
```
Comment (felangel) [inline .github/workflows/wish_deposit.yaml]:
```suggestion
name: wish_deposit
```

**Review by felangel — DISMISSED** [review]

**Review by felangel — APPROVED** [review]
